RFS/src/main/java/com/rfs/common/RestClient.java

```diff
@@ -1,18 +1,49 @@
 package com.rfs.common;
 
 import java.util.Base64;
+import java.util.function.BiConsumer;
+import java.util.function.IntConsumer;
 
+import com.rfs.netty.ReadMeteringHandler;
+import com.rfs.netty.WriteMeteringHandler;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.instrumentation.annotations.SpanAttribute;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
 import reactor.core.publisher.Mono;
 import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.client.HttpClientRequest;
 import reactor.netty.ByteBufMono;
+import reactor.netty.Connection;
 
 public class RestClient {
+
+    private static final AttributeKey<String> ATTRIBUTE_HTTP_METHOD = AttributeKey.stringKey("HttpMethod");
+    private static final AttributeKey<String> ATTRIBUTE_PATH = AttributeKey.stringKey("Path");
+    private static final LongCounter readBytesCounter;
+    private static final LongCounter writeBytesCounter;
+    static {
+        final var meter = GlobalOpenTelemetry.getMeter("RestClient");
+        readBytesCounter = meter.counterBuilder("readBytes")
+            .setDescription("Counts the number of bytes read")
+            .setUnit("1")
+            .build();
+        writeBytesCounter = meter.counterBuilder("writeBytes")
+            .setDescription("Counts the number of bytes written")
+            .setUnit("1")
+            .build();
+    }
+
     public static class Response {
         public final int code;
         public final String body;
         public final String message;
 
-        public Response(int responseCode, String responseBody, String responseMessage) {
+        @WithSpan
+        public Response(@SpanAttribute("responseCode") int responseCode, String responseBody, String responseMessage) {
```

Comment:
Ooh good callout - I missed that this was a constructor.  I'm guessing it would only track until the constructor returned the instance.

_Deferring since we aren't going to move forward with automatic instrumentation_